src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -272,6 +272,8 @@ type internal CompilerCaches(sizeFactor: int) =
 
     member val ParseAndCheckProject = AsyncMemoize(sf, 2 * sf, name = "ParseAndCheckProject")
 
+    member val RecentCheckFileResults: AsyncMemoize<_, int, _> = AsyncMemoize(sf, 2 * sf, name = "RecentCheckFileResults")
```

Comment:
Do we really need a separate cache for this? I thought we should just look into `ParseAndCheckFileInProject` and take the latest result from there if it's available.